torch/core

```diff
@@ -0,0 +1 @@
+..
```

Comment:
@ceriottm @frostedoyster @nickjbrowning bringing your attention to this. The `torch/core` folder is actually a symlink back to the root of the repo, allowing to include this code inside the sphericart_torch sdist. It might be a bit confusing though.

This is the cleanest way I found to be able to both build the torch bindings from the root project (for C++ workflows) and packaging the core source files in the source distribution of the sphericart_torch Python package, but if you have better ideas I'm listening!